src/uct/ib/base/ib_device.c

```diff
@@ -983,6 +983,76 @@ ucs_status_t uct_ib_device_port_check(uct_ib_device_t *dev, uint8_t port_num,
     return UCS_OK;
 }
 
+ucs_status_t
+uct_ib_device_ece_check(uct_ib_device_t *dev,
+                        struct ibv_context *ctx, struct ibv_pd *pd)
+{
+    ucs_status_t status = UCS_OK;
+#ifdef HAVE_DECL_IBV_ECE
+    struct ibv_ece ece  = {};
+    struct ibv_qp *dummy_qp;
+    struct ibv_cq *cq;
+    struct ibv_qp_init_attr qp_init_attr;
+
+    cq = ibv_create_cq(ctx, 1, NULL, NULL, 0);
+    if (cq == NULL) {
+        status = UCS_ERR_IO_ERROR;
+        goto out;
+    }
+
+    memset(&qp_init_attr, 0, sizeof(qp_init_attr));
+    qp_init_attr.send_cq = cq;
+    qp_init_attr.recv_cq = cq;
+    qp_init_attr.qp_type = IBV_QPT_RC;
+
+    qp_init_attr.cap.max_send_wr  = 1;
+    qp_init_attr.cap.max_recv_wr  = 1;
+    qp_init_attr.cap.max_send_sge = 1;
+    qp_init_attr.cap.max_recv_sge = 1;
+
+    dummy_qp = ibv_create_qp(pd, &qp_init_attr);
+    if (dummy_qp == NULL) {
+        status = UCS_ERR_IO_ERROR;
+        goto free_cq;
+    }
+
+    /* ibv_set_ece check whether ECE is supported */
+    if ((ibv_query_ece(dummy_qp, &ece) == 0) &&
+        (ibv_set_ece(dummy_qp, &ece) == 0)) {
+        dev->flags |= UCT_IB_DEVICE_FLAG_ECE;
+    }
+
+    ibv_destroy_qp(dummy_qp);
+free_cq:
+    ibv_destroy_cq(cq);
+out:
+#endif
+    return status;
+}
+
+ucs_status_t
+uct_ib_set_ece(uct_ib_device_t *dev, struct ibv_qp *qp, uint32_t ece_val)
+{
+    struct ibv_ece UCS_V_UNUSED ece;
```

Comment:
why UCS_V_UNUSED?

Reply:
@yosefe 
According to below macro definition
```
#define ibv_query_ece(...)  ({ errno = EOPNOTSUPP; UCS_ERR_UNSUPPORTED; })
#define ibv_set_ece(...)    ({ errno = EOPNOTSUPP; UCS_ERR_UNSUPPORTED; }
```
The variable ```ece``` here maybe not used after compiler optimization even ```ece.options = ece_val;``` is wrote in this function. Or, the CI will report the error like that ```ece is defined but not used```

Reply:
Done